build/common.rs

```diff
@@ -457,8 +457,15 @@ pub fn setup_clang_env(path: Option<&Path>) -> Result<()> {
 
         if policy != "ignore" {
             #[allow(deprecated)]
-            let espup_clang_path =
-                std::env::home_dir().map(|home| home.join(".espup").join("esp-clang"));
+            let espup_clang_path = std::env::home_dir().map(|home| {
+                const ESPUP_CLANG_SUFFIX: &str = "espup/esp-clang";
+                let local_share_path = home.join(".local").join("share").join(ESPUP_CLANG_SUFFIX);
```

Comment:
While local share is the "default xdg data home" its not really a "xdg support" per se, For that we would have to respect the respective xdg env var itself. 

I would not be opposed to use it if its set by the user. But crucially the change should work not only for linux people, but should not change the default for windows and mac users. 

Site note:  keep in mind that this particular case where we are "using the espup" provided esp-clang version we are in a separate file path looking. There is a certain risk to merge the espclang path into a directory where users might have other clang versions installed and may then provide wrong versions. E.g the user provided install overrides the espup install. By having the default in a separate folder its less risk.